From 7b25b99be3b1bc0e70488584375113fa47a65a6d Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Wed, 28 Apr 2021 19:32:44 +0800
Subject: [PATCH 04/13] drivers: pci: s32gen1: fix 's32gen1_pcie_msi_handler'
 defined but not used warning

The s32gen1_pcie_msi_handler() is referenced by s32gen1_pcie_config_irq()
when CONFIG_PCI_S32GEN1_EP_MSI kernel config is enabled, but currentlly,
this config is disabled, so add if-conditions to fix the following warning:

	warning: 's32gen1_pcie_msi_handler' defined but not used [-Wunused-function]
	702 | static irqreturn_t s32gen1_pcie_msi_handler(int irq, void *arg)

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>

---
 drivers/pci/controller/dwc/pci-s32gen1.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/controller/dwc/pci-s32gen1.c b/drivers/pci/controller/dwc/pci-s32gen1.c
index eec9f873c2a6..0d32cf40cbce 100644
--- a/drivers/pci/controller/dwc/pci-s32gen1.c
+++ b/drivers/pci/controller/dwc/pci-s32gen1.c
@@ -671,7 +671,7 @@ static void s32gen1_pcie_stop_link(struct dw_pcie *pcie)
 	s32gen1_pcie_disable_ltssm(s32_pp);
 }
 
-#ifdef CONFIG_PCI_MSI
+#ifdef CONFIG_PCI_S32GEN1_EP_MSI
 /* msi IRQ handler
  * irq - interrupt number
  * arg - pointer to the "struct pcie_port" object
@@ -688,7 +688,7 @@ static irqreturn_t s32gen1_pcie_msi_handler(int irq, void *arg)
 
 	return dw_handle_msi_irq(pp);
 }
-#endif
+#endif /* CONFIG_PCI_S32GEN1_EP_MSI */
 
 static int s32gen1_pcie_host_init(struct pcie_port *pp)
 {
-- 
2.25.1

